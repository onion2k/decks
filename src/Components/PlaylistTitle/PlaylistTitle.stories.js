import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import PlaylistTitle from '../Components/PlaylistTitle';

storiesOf('PlaylistTitle', module)
  .add('Basic', () => (
    <PlaylistTitle />
  ));
