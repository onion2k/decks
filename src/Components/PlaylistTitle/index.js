import React, { Component } from 'react';

class PlaylistTitle extends Component {

    render() {

        return (
            <div className="PlaylistTitle">
                <span>{ this.props.title }</span>
                <button>Playlists</button>
            </div>
        );
    }
}

export default PlaylistTitle;