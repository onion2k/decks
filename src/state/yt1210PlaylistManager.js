import { configure, decorate, observable, action, computed } from "mobx";

configure({ enforceActions: true });

export default class yt12010PlaylistManager {

    // Playlists

    playlist = 1;
    track = 0;
    currentTrack = { videoId: "", label: "" };

    playlists = [
        { "id": 1, "title": "Playlist 1", tracks: [
          { title: "", videoId: "BuVJEn9wk9Y", duration: "" },
          { title: "", videoId: "Rs38lKxmtI4", duration: "" },
          { title: "", videoId: "v_yTphvyiPU", duration: "" },
          { title: "", videoId: "e4Ao-iNPPUc", duration: "" },
          { title: "", videoId: "FALYmqt-7TQ", duration: "" },
          { title: "", videoId: "a3Z4RWZa9WA", duration: "" }
        ]},
        { "id": 2, "title": "Playlist 2", tracks: [
            { title: "", videoId: "a3Z4RWZa9WA", duration: "" },
            { title: "", videoId: "BuVJEn9wk9Y", duration: "" },
            { title: "", videoId: "Rs38lKxmtI4", duration: "" },
            { title: "", videoId: "v_yTphvyiPU", duration: "" },
            { title: "", videoId: "e4Ao-iNPPUc", duration: "" },
            { title: "", videoId: "FALYmqt-7TQ", duration: "" }
          ]}
    ];

    load = (id) => {
        this.playlist = id;
        // const pl = this.playlists.find((playlist) => playlist.id === this.playlist).tracks;
    };

    // add playlist
    playlistAdd = (title)=>{
        this.playlists.push({ id: this.playlists.length+1, title: title, tracks: [] });
    };

    // delete playlist
    playlistDelete = (index)=>{
        this.playlists.splice(index, 1);
    };

    // add track
    playlistAddTrack = (id, videoId)=>{
        this.playlist = this.playlists.find((playlist) => playlist.id === id);
        this.playlist.tracks.push({ title: "", videoId: videoId, duration: "" });
    };

    // delete track
    playlistDeleteTrack = (id, index)=>{
        this.playlist = this.playlists.find((playlist) => playlist.id === id);
        this.playlist.tracks.splice(index, 1);
    };

    get tracks() {
        return this.playlists.find((playlist) => playlist.id === this.playlist).tracks;
    };

    updateTrackData = (data, duration) => {
        const track = this.tracks.find((track)=>track.videoId === data.video_id);
              track.title = data.title;
              track.duration = duration;
        this.currentTrack = track;
    };

    getTracks = () => {
        return this.tracks;
    };

    getCurrentTrack = () => {
        return this.tracks[this.track];
    };

    getNextTrack = () => {
        return this.tracks[this.track++];
    };

    getTrackById = (id) => {
        this.track = this.tracks.findIndex((track) => track.videoId === id);
        return this.tracks[this.track];
    };
}

decorate(yt12010PlaylistManager, {
    currentTrack: observable,
    playlists: observable,
    playlist: observable,
    track: observable,
    tracks: computed,
    load: action,
    playlistAdd: action,
    playlistDelete: action,
    playlistAddTrack: action,
    playlistDeleteTrack: action,
    updateTrackData: action,
    getCurrentTrack: action,
    getNextTrack: action
});
