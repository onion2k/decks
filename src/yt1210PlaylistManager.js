import { configure, decorate, observable, action, autorun } from "mobx";

export default class yt12010PlaylistManager {

    // Playlists

    playlist = [];

    playlists = [
        { "id": 1, "title": "Playlist 1", tracks: [
          { title: "", videoId: "BuVJEn9wk9Y", duration: "" },
          { title: "", videoId: "Rs38lKxmtI4", duration: "" },
          { title: "", videoId: "v_yTphvyiPU", duration: "" },
          { title: "", videoId: "e4Ao-iNPPUc", duration: "" },
          { title: "", videoId: "FALYmqt-7TQ", duration: "" },
          { title: "", videoId: "a3Z4RWZa9WA", duration: "" }
        ]},
        { "id": 2, "title": "Playlist 2", tracks: [0,1,2,3,4,5] },
        { "id": 3, "title": "Playlist 3", tracks: [0,1,2,3,4,5] },
        { "id": 4, "title": "Playlist 4", tracks: [0,1,2,3,4,5] }
    ];

    load = action((id) => {
        this.playlist = this.playlists.find((playlist) => playlist.id === id).tracks;
        this.videoId = this.playlist[0].videoId;
        this.playing = true;
    });

    // add playlist
    playlistAdd = action((title)=>{
        this.playlists.push({ id: this.playlists.length+1, title: title, tracks: [] });
    });

    // delete playlist
    playlistDelete = action((index)=>{
        this.playlists.splice(index, 1);
    });

    // add track
    playlistAddTrack = action((id, videoId)=>{
        this.playlist = this.playlists.find((playlist) => playlist.id === id);
        this.playlist.tracks.push({ title: "", videoId: videoId, duration: "" });
    });

    // delete track
    playlistDeleteTrack = action((id, index)=>{
        this.playlist = this.playlists.find((playlist) => playlist.id === id);
        this.playlist.tracks.splice(index, 1);
    });

    updateTrackData = action((data, duration) => {

    });

}

decorate(yt12010PlaylistManager, {
    playlist: observable,
    playlists: observable,
    load: action,
    playlistAdd: action,
    playlistDelete: action,
    playlistAddTrack: action,
    playlistDeleteTrack: action,
    updateTrackData: action
});