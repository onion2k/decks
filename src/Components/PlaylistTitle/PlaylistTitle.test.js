import React from "react";
import renderer from "react-test-renderer";
import { MemoryRouter } from 'react-router-dom'

import PlaylistTitle from ".";

import yt1210PlaylistManager from "../../state/yt1210PlaylistManager.js"
const playlistManagerState = new yt1210PlaylistManager();

describe("Playlist Title", () => {
  const onClick = jest.fn();

  it("renders properly", () => {
    const playlistTitle = renderer
      .create(
        <PlaylistTitle playlistManager={playlistManagerState} />
      )
      .toJSON();
    expect(playlistTitle).toMatchSnapshot();
  });
});
