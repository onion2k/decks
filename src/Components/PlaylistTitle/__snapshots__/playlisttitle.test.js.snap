// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist Title renders properly 1`] = `
<div
  className="PlaylistTitle"
>
  <span>
    Playlist 1
  </span>
</div>
`;
